org.adempiere.base/src/org/compiere/model/MProcessDrillRule.java

```diff
@@ -0,0 +1,212 @@
+		if (retValue.get_ID () == AD_Process_DrillRule_ID)
+		{
+			s_cache.put (AD_Process_DrillRule_ID, retValue, e -> new MProcessDrillRule(Env.getCtx(), e));
+			return retValue;
+		}
+		return null;
+	}
+
+	
+	public MProcessDrillRule(Properties ctx, int AD_Process_DrillRule_ID, String trxName) {
+		super(ctx, AD_Process_DrillRule_ID, trxName);
+	}
+
+	public MProcessDrillRule(Properties ctx, int AD_Process_DrillRule_ID, String trxName, String[] virtualColumns) {
+		super(ctx, AD_Process_DrillRule_ID, trxName, virtualColumns);
+	}
+
+	public MProcessDrillRule(Properties ctx, ResultSet rs, String trxName) {
+		super(ctx, rs, trxName);
+	}
+	
+	/**
+	 *
+	 * @param copy
+	 */
+	public MProcessDrillRule(MProcessDrillRule copy)
+	{
+		this(Env.getCtx(), copy);
+	}
+
+	/**
+	 *
+	 * @param ctx
+	 * @param copy
+	 */
+	public MProcessDrillRule(Properties ctx, MProcessDrillRule copy)
+	{
+		this(ctx, copy, (String) null);
+	}
+
+	/**
+	 *
+	 * @param ctx
+	 * @param copy
+	 * @param trxName
+	 */
+	public MProcessDrillRule(Properties ctx, MProcessDrillRule copy, String trxName)
+	{
+		this(ctx, 0, trxName);
+		copyPO(copy);
+	}
+
+	/**
+	 * Get array of MProcessDrillRule by Table
+	 * @param ctx
+	 * @param AD_Table_ID
+	 * @param trxName
+	 * @return
+	 */
+	public static MProcessDrillRule[] getByTable(Properties ctx, int AD_Table_ID, String trxName) {
+
+		String whereClause = " AD_Table_ID = ? ";
+		List<MProcessDrillRule> processDrillRules = new Query(ctx, MProcessDrillRule.Table_Name, whereClause, trxName)
+				.setParameters(AD_Table_ID)
+				.setClient_ID()
+				.setOnlyActiveRecords(true)
+				.list();
+
+		return processDrillRules.toArray(new MProcessDrillRule[processDrillRules.size()]);
+	}
+
+	/**
+	 * Get array of MProcessDrillRule by Column Name
+	 * @param ctx
+	 * @param columnName
+	 * @param trxName
+	 * @return
+	 */
+	public static MProcessDrillRule[] getByColumnName(Properties ctx, String columnName, String trxName) {
+
+		String whereClause = "";
+		List<MProcessDrillRule> processDrillRules = new Query(ctx, MProcessDrillRule.Table_Name, whereClause, trxName)
+				.addJoinClause(" INNER JOIN AD_Column c ON "+ MProcessDrillRule.Table_Name + ".AD_Column_ID = c.AD_Column_ID AND c." + MColumn.COLUMNNAME_ColumnName + " = ?")
+				.setParameters(columnName)
+				.setClient_ID()
+				.setOnlyActiveRecords(true)
+				.list();
+
+		return processDrillRules.toArray(new MProcessDrillRule[processDrillRules.size()]);
+	}
+
+	@Override
+	protected boolean beforeSave(boolean newRecord) {
+
+		if(newRecord || is_ValueChanged(MProcessDrillRule.COLUMNNAME_AD_Process_ID)) {
+			MProcess process = MProcess.get(getAD_Process_ID());
+			if(process != null && process.getAD_ReportView_ID() > 0) {
+				MReportView reportView = MReportView.get(process.getAD_ReportView_ID());
+				if(reportView != null && reportView.getAD_Table_ID() > 0)
+					setAD_Table_ID(reportView.getAD_Table_ID());
```

Comment:
Why are we setting the table field to the ad_table_id of report view here (same thing in the callout code above) ? This wasn't the case with https://wiki.idempiere.org/en/NF10_DrillAssistant where table field is set to M_Product. Please clarify if the example given in the wiki page is wrong and there's a different prefer usage flow here.